src/classes/CRLP_RollupQueryBuilder.cls

```diff
@@ -0,0 +1,312 @@
+                if (!String.isEmpty(addlFilter)) {
+                    whereClauses.add(addlFilter);
+                }
+            }
+
+            // Exclude Organization Donations from Contact Hard Credit rollups (where Opp.Account.SYSTEM_IsIndividual=false)
+            // When the npo02__Always_Rollup_to_Primary_Contact__c is field is unchecked.
+            if (summaryObjectType == Contact.SObjectType && householdSettings.npo02__Always_Rollup_to_Primary_Contact__c == false) {
+                whereClauses.add('Account.npe01__SYSTEMIsIndividual__c = True');
+            }
+
+            // Force the details query to sort by Ascending CreatedDate so that the records are process
+            // in the order in which they were created in the database. Thus the 'first' is truly the first
+            // and the last is truly the last.
+            orderByFields.add('CreatedDate ASC');
+        }
+
+        if (!whereClauses.isEmpty()) {
+            query += ' WHERE ' + String.join(whereClauses, ' AND ');
+        }
+        if (customWhereClauses != null && !customWhereClauses.isEmpty()) {
+            query += ' WHERE ' + String.join(customWhereClauses, ' AND ');
+        }
+        if (!orderByFields.isEmpty()) {
+            query += ' ORDER BY ' + String.join(orderByFields, ', ');
+        }
+
+        return query;
+    }
+
+    /**
+     * @description Returns the SOQL string for a Query against the details object. This is used for a NonSkew job
+     * or for the Queueable rollup operation.
+     * DetailRecordId's are required to generate this query.
+     * @return SOQL String
+     */
+    public String getRelatedRecordsQueryAsSOQL() {
+        String query = '';
+
+        return query;
+    }
+
+    /**
+     * @description Returns the SOQL string for a QueryLocator query for the specified Job Type (Skew or NonSkew)
+     * and RollupType. If there are SummaryRecordId's present, the query is filtered for those specified Id's.
+     * @return SOQL String
+     */
+    /*public String getQueryLocatorAsSOQL() {
+        String query = '';
+
+        return query;
+    }*/
+
+    /**
+     * @description Build a complete SOQL query string from a set of Summary Record Ids, including the requried
+     * child objects and filtered by the parent record id's
+     * @return SOQL
+     */
+    private String buildFullDetailsQueryForSummaryRecords() {
+        SObjectType mainObject = Opportunity.SObjectType;
+        String detailKeyField;
+        List<SObjectType> childObjects = new List<SObjectType>();
+
+        SObjectType summaryObjectType = CRLP_Rollup_SVC.getSummaryObjectType(rollupType);
+
+        // The specific set of 'detail' records is fixed based on the target summary object.
+        if (summaryObjectType == Account.SObjectType) {
+
+            if (rollupType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
+                || rollupType == CRLP_RollupProcessingOptions.RollupType.AccountContactSoftCredit
```

Comment:
The method is called for hard credits, but the line above has the check for the Account Soft Credit. is additional refactoring required?

Reply:
Hmm. 🤷‍♂️

Reply:
It is good that refactoring is done while all tests are intact. Probably further refactoring and code extracting (if any) will be done when this class is completely tested on its own.